File: src/main/scala/dev/tlang/tlang/tmpl/TmplNode.scala
package dev.tlang.tlang.tmpl

import dev.tlang.tlang.ast.common.value.EntityValue
import dev.tlang.tlang.ast.model.set.ModelSetEntity
import tlang.core
import tlang.internal.DeepCopy

trait TmplNode[T] extends core.Value[T] with DeepCopy {

  def toEntity: EntityValue

  def toModel: ModelSetEntity
}

object TmplNode {
  val name: String = this.getClass.getSimpleName.replace("$", "")
}

File: src/main/java/tlang/internal/TmplNode.java
package tlang.internal;

import tlang.core.Entity;
import tlang.core.Type;

public interface TmplNode<T> extends Element<T>, Context {

    Type TYPE = ClassType.of(TmplNode.class);

    String name = TmplNode.class.getSimpleName();

    Entity toEntity();

//    Model toModel();

}
